### ISSUE TITLE
dem_create hangs when using gdal 3.4.1 and pyroSAR 0.16

### ISSUE THREAD

**Issue by peterfriedl** [johntruckenbrodt/pyroSAR#200]
If using the latest version of pyroSAR (0.16), dem_create hangs when trying to mosaic a DEM from a VRT built with the latest version of gdal (3.4.1). No error message appears. Executing the same commands using gdal 3.3 and pyroSAR 15.1 works fine: 

```
search = cat.search(
    collections = "COPERNICUS_DEM_1ARC_COG",
    bbox=[lon1, lat1, lon2, lat2]
)

items = search.get_all_items()
item_json = items.to_dict()

tiles_to_mosaic=[]

for i in range (len(items)):
   tile=item_json['features'][i]['assets']['data']['href']
   tiles_to_mosaic.append(tile)

vrt=gdal.BuildVRT(outdir + "/dem_tmp.vrt", tiles_to_mosaic)

dem_create(src=vrt, dst=outdir + "/dem.tif", geoid_convert=True, geoid="EGM2008")
```

**Comment by johntruckenbrodt** [comment]
Thanks @peterfriedl. I will investigate this next week. In the meantime you can make use of the new arguments `pbar` and `threads` to check the status and (if it doesn't hang but is just slow) increase the speed.

**Comment by peterfriedl** [comment]
Thanks @johntruckenbrodt for the hint to the `pbar` and `threads` arguments. Indeed, `dem_create` does not hang, it's just super slow when using its default configuration (a few seconds with pyroSAR 0.15.1 and gdal 3.3 vs. > 11 minutes with pyroSAR 0.16 and gdal 3.4.1). I tried different numbers of 56 available threads and found the "sweet spot" of our system to be at only 2 threads. Adding more threads made things even worse by leading to execution times of up to > 30 minutes. Using `threads=2` reduced the execution time to 7 minutes, but this is still far away from the previous execution times of just a few seconds, achieved with the default configurations of `dem_create.`

**Comment by johntruckenbrodt** [comment]
Thanks @peterfriedl, this is super valuable feedback. I did notice `dem_create` taking quite long, but thought it was just due to the large 10 m DEM mosaics I was creating at the time. The configuration of 0.15.1 was to use multithreading but leave the `GDAL_NUM_THREADS` variable untouched. I introduced the `threads` argument because it was so slow. I suspect this could have something to do with the GDAL version. Would you mind checking this? You could create a conda env like so:
```
conda create --name test pyroSAR gdal=3.3
```

**Comment by peterfriedl** [comment]
@johntruckenbrodt I can confirm that the problem is definitely caused by gdal. Executing dem_create in pyroSAR 0.16 (default configuration) with gdal 3.3.3 just takes 8 seconds for the test dataset - with gdal 3.4.1 it takes > 11 minutes.

**Comment by peterfriedl** [comment]
@johntruckenbrodt I now figured out that my initial issue of dem_create hanging is still true. I executed the software (with gdal 3.4.1) in a container on different nodes on an HPC-cluster. I observed that sometimes dem_create is just very slow but sometimes it really hangs.

**Comment by peterfriedl** [comment]
@johntruckenbrodt I am not sure whether it has something to do with the gdal-version in general. I tried both: a) gdal 3.3.3 + pyroSAR 0.16 and b) gdal 3.4.1 + pyroSAR 0.16. in case of b) dem_create runs super slow and in case of a) dem-create hangs.

**Comment by johntruckenbrodt** [comment]
If it hangs then this might be due to function [get_egm_lookup](https://pyrosar.readthedocs.io/en/latest/pyroSAR.html#pyroSAR.auxdata.get_egm_lookup). However, there are definitely strange things going on. I have opened a new GDAL issue: https://github.com/OSGeo/gdal/issues/5414.

**Comment by johntruckenbrodt** [comment]
Okay after the discussion in the GDAL issue above pyroSAR is now using different EGM lookup files, see https://github.com/johntruckenbrodt/pyroSAR/commit/598ac39af7ffc303777e6d10b931f8d7788afbae.
@peterfriedl, I will publish pyroSAR version 0.16.1 later. Perhaps this already fixes this issue.
Also interesting in this context: https://github.com/OSGeo/gdal/issues/3630.

**Comment by peterfriedl** [comment]
@johntruckenbrodt thanks a lot for fixing this! Using all available threads, with version 0.16.1 the execution time of dem_create has been reduced to just 2 seconds. :-)
